draft-deepspace-ip-architecture.xml

```diff
@@ -216,6 +216,23 @@
+        <ul>
+          <li><t>Protocol Negotiation / "Happy Eyeballs" - Due to presence of both IPv4 and IPv6 in the terrestrial Internet, applications or transport stacks may include "happy eyeballs" capabilities <xref target="RFC8305"/> that make use of multiple DNS queries and connection attempts for different addresses and address families returned.  Downsides to this in interplanetary applications might include (1) the additional load on interplanetary DNS, (2) unnecessary use of network capacity for paralel connection attempts (e.g. if 0-RTT data transmission is used), and (3) unnecessary additional transport state maintained for an extended period of time.  If applications for interplanetary use initially rely on either fixed IP addresses (instead of DNS names) or a single address family (e.g. IPv6), then there should be no downsides to presence of happy eyeballs algorithms within the transport stack.  If happy eyeballs is present and triggered, then the values in section 8 of <xref target="RFC8305"/> should be carefully considered and tuned based on the DNS and path expectations (e.g. resolution delay, first address family count, connection attmept delay, minimum connection attempt delay, and maximum connection attempt delay) as most of these values are not likely to be appropriate even for Earth-lunar paths.</t></li>
+          <li><t>Connection Initiation / Handshaking - Due to the need to be robust to stale packets, errors, and denial of service attacks, Internet transports include handshaking state machines, such as 3-way and 4-way handshakes for connection establishment (the case can be even worse for some transport protocol and TLS combinations).  Since the interplanetary round trip times may be larger than the duration of contact periods, these handshaking mechanisms are very inefficient.  Though they are tolerable in cases such as between Earth and lunar networks, they are stifling for Earth-Mars and other interplanetary network paths.  Transports, such as QUIC, that have the ability to resume based on shared state from prior application connections or to rapidly start transferring data ("0RTT") can be suitably efficient, once initial state is obtained; however, they still require a complete initial handshake with the full set of round trip times imposed.  For interplanetary use, it may be beneficial to find ways to securely pre-set information to allow this more efficient startup, without requiring the full initial handshake even once.</t></li>
```

Comment:
We previously mentioned initiating the connection prior to initial launch.

Reply:
Definitely.   We can do that in many cases, though not 100% all.  Two (admittedly uncommon) examples would be:
(1) Science missions can complete out their designed-for mission, and then if the hardware is still operational, become repurposed for other uses.  This has happened before, and one of the discussed possibilities, for instance, is to have some science landers later become network basestations, if they are able to survive past completion of their primary mission.  In that case, the data flows they are a part of can change from what they were designed for at launch.
(2) Endpoints can change hands, either between public and private sector, or between different companies that merge or split over time, and the mission operations endpoints on earth may change.

There are likely some other cases where at-launch pre-configuration may not be fully viable.

Maybe these are rare enough compared to the nominal data flows that it would be silly to design anything specifically to accommodate them, but we should make agree such events aren't precluded at least, so ways to seed new connection state through management may be sensible.